Python/bytecodes.c

```diff
@@ -3961,12 +3961,19 @@ dummy_func(
         }
 
         op(_SAVE_CURRENT_IP, (--)) {
-            TIER_ONE_ONLY
+            #if TIER_ONE
             assert(frame->next_instr_offset == 0);
             frame->next_instr_offset = (uint16_t)(next_instr - frame->instr_ptr);
+            #endif
+            #if TIER_TWO
+            frame->next_instr_offset = oparg;
+            #endif
+            assert(frame->next_instr_offset != 0);
```

Comment:
Is this true for a JUMP_BACKWARD with offset 0?  

```
>>> def f():
...   while True: pass
... 
>>> dis.dis(f)
  1           0 RESUME                   0

  2     >>    2 JUMP_BACKWARD            2 (to 2)
```

Reply:
Oh, interesting counterexample. In fact, it seems this code gets the translator in a loop, repeating two uops forever:
```
  ADD_TO_TRACE(_JUMP_TO_TOP, 0, 0)
  ADD_TO_TRACE(_SET_IP, 1, 0)
```
I'll have to see if that's new or existing.

Reply:
That's not a new bug -- I can trigger it in main as well. So it's a good find! Until I fix it, it's immaterial whether that `assert` would fail, since such code can't be translated. But I'll remove it, since it serves no real purpose except to reveal my misunderstanding. :-)

Reply:
Oh wait. That's not really a bug, I didn't set the debug output level high enough. The translator is fine; it stops after adding 2 uops and then the trace is rejected because it's too short. The repetitive output is because we don't (yet) have exponential back-off in the translator -- it tries optimizing every 20 executions of the `JUMP_BACKWARD`, getting the same outcome each time.

I'll just push a commit that removes the `assert`, even though no trace should ever exhibit that problem.